easybuild/easyconfigs/h/HIP/HIP-5.6.0-GCCcore-11.3.0-amd.eb

```diff
@@ -0,0 +1,120 @@
+easyblock = 'CMakeMake'
+
+name = 'HIP'
+version = '5.6.0'
+versionsuffix = '-amd'
+
+homepage = 'https://github.com/ROCm-Developer-Tools/HIP'
+description = """HIP is a C++ Runtime API and Kernel Language that allows
+developers to create portable applications for AMD and NVIDIA GPUs from single
+source code."""
+docurls = ['https://rocmdocs.amd.com']
+
+toolchain = {'name': 'GCCcore', 'version': '11.3.0'}
+
+sources = [
+    {
+        'source_urls': ['https://github.com/ROCm-Developer-Tools/clr/archive/'],
+        'download_filename': 'rocm-%(version)s.tar.gz',
+        'filename': 'clr-rocm-%(version)s.tar.gz',
+    },
+    {
+        'source_urls': ['https://github.com/ROCm-Developer-Tools/HIP/archive/'],
+        'download_filename': 'rocm-%(version)s.tar.gz',
+        'filename': 'HIP-rocm-%(version)s.tar.gz',
+    },
+    {
+        'source_urls': ['https://github.com/ROCm-Developer-Tools/HIPCC/archive/'],
+        'download_filename': 'rocm-%(version)s.tar.gz',
+        'filename': 'HIPCC-rocm-%(version)s.tar.gz',
+    },
+]
+patches = [
+    'clr-rocm-5.6.0_fix_bad_message.patch',
+]
+checksums = [
+    {'clr-rocm-5.6.0.tar.gz': '8dcd99110737a294f67a805639cf372890c8ca16c7603caaa793e71e84478fe4'},
+    {'HIP-rocm-5.6.0.tar.gz': 'a8237768c1ae70029d972376f8d279f4de18a1e6106fff6215d1e16847bc375e'},
+    {'HIPCC-rocm-5.6.0.tar.gz': 'fdb7fdc9e4648376120330f034ee8353038d34c8a015f9eb0c208c56eeddd097'},
+    {'clr-rocm-5.6.0_fix_bad_message.patch': '99ef99d59ddfc1b46f243221fde98db18e5235df297e74badea2dfa3bbdb6d39'},
+]
+
+builddependencies = [
+    ('binutils', '2.38'),
+    ('CMake', '3.23.1'),
+    ('CppHeaderParser', '2.7.4'),
+]
+
+dependencies = [
+    ('zlib', '1.2.12'),
+    ('libxml2', '2.9.13'),
+    ('numactl', '2.0.14'),
+    ('Perl', '5.34.1'),
+    ('libglvnd', '1.4.0'),
+    ('Clang-AOMP', '%(version)s'),
+    ('ROCm-CompilerSupport', '%(version)s'),
+    ('rocm-cmake', '%(version)s'),
+    ('rocminfo', '%(version)s'),
+]
+
+# AMD uses 'gfx' to identify the GPU, the first number identifies the generation, according to
+# https://www.x.org/wiki/RadeonFeature/#index5h2 while the rest identifies the specific GPU.
+# In the context of EasyBuild this identifier can be thought of as equivalent to the 'sm_<xx>'
+# nomenclature of Nvidia.
+local_default_gfx = 'gfx908'
+
+# Need to build HIPCC first, in later versions it might be possible to make HIPCC a "standalone" package
+preconfigopts = "mkdir -p %(builddir)s/HIPCC-rocm-%(version)s/build; pushd %(builddir)s/HIPCC-rocm-%(version)s/build;"
+preconfigopts += " cmake ..; make; popd; echo HIPCC compiled; "
+
+_clopts = [
+    "-DHIP_COMMON_DIR=%(builddir)s/HIP-rocm-%(version)s/",
+    "-DHIP_PLATFORM=amd",
+    "-DHIPCC_BIN_DIR=%(builddir)s/HIPCC-rocm-%(version)s/build",
+    "-DHIP_CATCH_TEST=0",
+    "-DCLR_BUILD_HIP=ON",
+    "-DCLR_BUILD_OCL=OFF",
+    "-DOFFLOAD_ARCH_STR='--offload-arch=%s'" % local_default_gfx,
+]
```

Comment:
Also this EasyConfig worked pretty well for me, except that also here resulting libraries like `libamdhip64.so` showed to be linked to system libraries (like `/lib64/libnuma.so.1`, `/lib64/libstdc++.so.6`, ...). Do you see this as well? From what I could tell this is due to stripping of rpath information in the install step. Adding an extra `"-DCMAKE_SKIP_RPATH=ON",` option here prevents this from happening.

Reply:
I'll take a look at both of these as soon as I find some time. I think there is something called Christmas coming up fairly soon...

Reply:
I see no trace of this once the module is loaded.

Did you run the ldd without the module loaded?

CMAKE_SKIP_RPATH is only set to ON in the cmakemake easyblock if CMake < 3.5.0 and build_option('rpath') is set. And for this one CMake is 3.23...

Reply:
I was indeed running ldd after loading the module. After having a closer look I also managed to resolve it via the following patch (so instead of introducing `-DCMAKE_SKIP_RPATH=ON` in the easyconfig):
```diff
--- clr-rocm-5.6.0/hipamd/CMakeLists.txt.orig   2024-01-09 12:13:38.465256148 +0100
+++ clr-rocm-5.6.0/hipamd/CMakeLists.txt        2024-01-09 12:18:05.463862147 +0100
@@ -41,7 +41,7 @@

 # required to add the right link to libhsa-runtime in install/lib path
 # CMAKE_PREFIX_PATH is used as rpath to search for libs outside HIP
-set(CMAKE_INSTALL_RPATH "${CMAKE_PREFIX_PATH}/${CMAKE_INSTALL_LIBDIR}")
+set(CMAKE_INSTALL_RPATH "${CMAKE_PREFIX_PATH}")
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)

 #############################
```
To me, this `set(CMAKE_INSTALL_RPATH "${CMAKE_PREFIX_PATH}/${CMAKE_INSTALL_LIBDIR}")` line looked strange because EasyBuild sets `${CMAKE_INSTALL_RPATH}` equal to a list of semicolon-separated software directories. So in this case appending something like "/lib" to it and using that as CMAKE_INSTALL_RPATH does not seem quite right. It appears that this eventually caused the following kind of runtime path modifications in the install step:
```
...
-- Installing: <snip>/2022a/software/HIP/5.6.0-GCCcore-11.3.0/lib64/libhiprtc-builtins.so.5.6.31061
-- Installing: <snip>/2022a/software/HIP/5.6.0-GCCcore-11.3.0/lib64/libhiprtc-builtins.so.5
-- Set runtime path of "<snip>/2022a/software/HIP/5.6.0-GCCcore-11.3.0/lib64/libhiprtc-builtins.so.5.6.31061" to "/lib64"
...
```
which then led to the installed libraries linking to system libraries.

At the moment I don't really have an idea why it is not occurring on your side. I suppose that you don't see `Set runtime path ... ` messages in the EasyBuild logfile? Could you perhaps share that logfile?

(disclaimer: my CMake knowledge is rather limited)

Reply:
So, with that patch there are no longer any "Set runtime path" messages and RPATH info in at least libamdhip64.so is empty, 
Now I'll try to rebuild it with --rpath set...

Reply:
Seems to do good things with --rpath too.